File: summer2021/unit1/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        result = []
        for i in range(left, right + 1):
            if self.isSelfDividingNumber(i):
                result.append(i)
        return result
    
    def isSelfDividingNumber(self, n: int) -> bool:
        i = n
        while i > 0:
            d = (int)(i % 10)
            if d == 0 or (n % d) != 0:
                return False
            i //= 10
        return True
            

File: 2026/week1/self-dividing-numbers.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> result;
        for (int i = left; i <= right; i++) {
            if (isSelfDividingNumber(i)) {
                result.push_back(i);
            }
        }
        return result;
    }

    bool isSelfDividingNumber(int num) {
        int current = num;
        int remainder;
        while (current) {
            remainder = current % 10;
            if (remainder == 0 || num % remainder != 0) {
                return false;
            }
            current = floor(current / 10);
        }
        return true;
    }
};
